src/preconditions/argument/HasMsg.js

```
@@ -17,39 +17,51 @@
  */
 "use strict";
 const {ArgumentPrecondition, PreconditionResult} = require("patron.js");
+const db = require("../../services/database.js");
+const message = require("../../utilities/message.js");
 const {data: {responses}} = require("../../services/data.js");
 const str = require("../../utilities/string.js");
-const time = require("../../utilities/time.js");
 
-module.exports = new class BetweenTime extends ArgumentPrecondition {
+module.exports = new class Between extends ArgumentPrecondition {
   constructor() {
-    super({name: "betweentime"});
+    super({name: "hasmsg"});
   }
 
-  async run(cmd, msg, arg, args, val, opt) {
-    if (val == null || ((opt.min == null || val >= opt.min) && (opt.max == null
-        || val <= opt.max)))
-      return PreconditionResult.fromSuccess();
+  async run(cmd, msg, arg, args, val) {
+    const ids = message.getIds(val);
+    const {user} = args;
+    let userSent = false;
 
-    if (opt.min == null) {
+    if (ids == null) {
       return PreconditionResult.fromError(
         cmd,
-        `the ${arg.name} must be below ${time.format(opt.max)}.`
+        str.format(responses.hasMsg, arg.name)
       );
     }
 
-    if (opt.max == null) {
+    for (let i = 0; i < ids.length; i++) {
+      const res = await db.getFirstRow(
+        "SELECT author_id FROM messages WHERE (guild_id, id) = ($1, $2)",
+        [msg.channel.guild.id, ids[i]]
+      );
+
+      if (res == null) {
+        return PreconditionResult.fromError(
+          cmd,
+          str.format(responses.hasMsg, arg.name)
+        );
+      } else if (res.author_id === user.id) {
+        userSent = true;
+      }
+    }
+
+    if (userSent === false) {
```

What is the point of this bool? Why not just return the precondition failure when you set `userSent` to true?

---

Because first off when it's true it doesn't return the failure, secondly because the goal is at least one message having been sent by the user, not all of them.